### PR TITLE
search highlights for search in Reader

### PR DESCRIPTION
**Pull request by tejans24**

### CODE REVIEW

File: random.html [hunk 1]
```diff
@@ -0,0 +1 @@
...2674586 L10.4497163,14.2184033 L17.1414226,19.2674586 C17.2700654,19.3683882 17.4215057,19.4229048 17.5791339,19.4258517 C17.8575886,19.4258517 18.2252791,19.371335 18.5268569,19.0431298 C18.8297375,18.7149247 18.8984556,18.2762127 18.8984556,17.9336416 L18.8984556,3.68789356 C18.8984556,2.16989858 17.6820481,1 16.2585095,1 L4.64092309,1 Z"></path></g></g></svg></li></ul></td><td class="receipt-date-column" colspan="1"><span class="document-list-receipt-date"><div aria-label="search result"><span><span class="">01/02/2017</span></span></div></span></td><td class="doc-type-column" colspan="1"><strong><a href="documents/1" aria-label="bva decision unopened"><div aria-label="search result"><span><span class="">bva decision</span></span></div></a></strong></td><td class="tags-column" colspan="1"><div class="document-list-issue-tags"><div class="document-list-issue-tag"><div aria-label="search result"><span><span class="">mytag</span></span></div></div></div></td><td class="comments-column" colspan="1"><span class="document-list-comments-indicator"></span></td></tr></tbody><tfoot></tfoot></table></div></div></div></div><!-- react-empty: 138 --></div><!-- react-empty: 13 --></div>ÏÏÏÏÏÏÏ
```
Comment (mdbenjam) [inline random.html]: What is this?

File: client/app/components/Comment.jsx [hunk 2]
```diff
@@ -86,7 +87,10 @@ export class Comment extends React.Component {
           key={this.props.children.toString()}
           id={this.props.id}
           onClick={this.onClick}>
-          {this.props.children}
+          <Highlight
+            searchQuery={searchQuery}
+            textToHighlight={this.props.children}
```
Comment (mdbenjam) [inline client/app/components/Comment.jsx]: Sort of a nit, but can we make textToHighlight this component's children? I think that's what we've been doing in general, and I think it makes the jsx easier to read.

File: client/app/components/Highlight.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    return <div aria-label="search result">
+      <Highlighter
+        searchWords={_.union([searchQuery], searchQuery.split(' '))}
+        textToHighlight={textToHighlight}
+      />
+    </div>;
+  }
+}
+
+Highlight.propTypes = {
```
Comment (mdbenjam) [inline client/app/components/Highlight.jsx]: Why not connect this highlight component directly to the store? Then we don't have to make so many other wrappers.

File: client/app/reader/reducer.js [hunk 4]
```diff
@@ -107,6 +62,77 @@ const getExpandAllState = (documents) => {
   return Boolean(allExpanded);
 };
 
+const updateFilteredDocIds = (nextState) => {
```
Comment (mdbenjam) [inline client/app/reader/reducer.js]: Can we move this code back up? We have a pretty large unnecessary diff here.

File: client/app/components/Highlight.jsx [hunk 5]
```diff
@@ -1,24 +1,30 @@
...
 
 Highlight.propTypes = {
-  searchQuery: PropTypes.string.isRequired,
-  textToHighlight: PropTypes.string.isRequired
+  searchQuery: PropTypes.string.isRequired
 };
 
-export default Highlight;
+const mapStateToProps = (state) => ({
+  searchQuery: state.ui.docFilterCriteria.searchQuery
```
Comment (mdbenjam) [inline client/app/components/Highlight.jsx]: We're still passing in searchQuery everywhere, even though we map it here. Can we remove that?

File: client/app/components/reader/TagTableColumn.jsx [hunk 6]
```diff
@@ -1,22 +1,33 @@
...
+              {tag.text}
+            </Highlight>
           </div>;
       })}
     </div>;
   }
 }
 
+const mapStateToProps = (state) => ({
+  searchQuery: state.ui.docFilterCriteria.searchQuery
```
Comment (mdbenjam) [inline client/app/components/reader/TagTableColumn.jsx]: Not necessary anymore

File: client/test/node/reader/DecisionReviewer-test.js [hunk 7]
```diff
@@ -387,6 +387,26 @@ describe('DecisionReviewer', () => {
         expect(textArray).to.have.length(0);
       });
 
+      it('does search highlighting for matched keywords', () => {
+        wrapper.find('input').simulate('change',
+          { target: { value: 'mytag form' } });
+
+        const doesArrayIncludeString = (array, string) => array.some((item) => item.includes(string));
```
Comment (mdbenjam) [inline client/test/node/reader/DecisionReviewer-test.js]: Might this be easier if we just look for the `mark` tag directly?

File: client/app/components/Highlight.jsx (hunk 5 above)
Comment (tejans24) [inline client/app/components/Highlight.jsx]: I thought I removed them, let me double check.

File: yarn.lock [hunk 8]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (mdbenjam) [inline yarn.lock]: Why are we committing a yarn file?

File: client/test/node/reader/DecisionReviewer-test.js [hunk 9]
```diff
@@ -387,6 +387,25 @@ describe('DecisionReviewer', () => {
...
+        let textArray = wrapper.find('mark').
+          map((node) => node.html());
```
Comment (mdbenjam) [inline client/test/node/reader/DecisionReviewer-test.js]: maybe look at text instead of html.

File: client/app/reader/DocumentsTable.jsx [hunk 10]
```diff
@@ -87,6 +88,35 @@ const lastReadIndicatorMapStateToProps = (state, ownProps) => ({
 });
 const ConnectedLastReadIndicator = connect(lastReadIndicatorMapStateToProps)(LastReadIndicator);
 
+class DocTypeColumn extends React.PureComponent {
```
Comment (mdbenjam) [inline client/app/reader/DocumentsTable.jsx]: Now that we're connecting highlight directly to the store, do we really need this class?

File: yarn.lock (hunk 8 above)
Comment (tejans24) [inline yarn.lock]: grrrrr....my fault. I'll keep an eye out for these

File: client/app/reader/DocumentsTable.jsx (hunk 10 above)
Comment (tejans24) [inline client/app/reader/DocumentsTable.jsx]: I thought there is enough logic for this component to worth keeping it separate.

File: client/test/node/reader/DecisionReviewer-test.js (hunk 9 above)
Comment (tejans24) [inline client/test/node/reader/DecisionReviewer-test.js]: Sure, any specific advantages of doing that?
